web-app/HeimScripts/heatmap/run.R

```diff
@@ -56,6 +57,42 @@ main <- function(max_rows=100){
   list(messages=msgs) # main function in every R script has to return a list (so a data.frame will also do)
 }
 
+verifyInput <- function(max_rows, sorting){
+  if(max_rows <= 0 ){
+    stop("Max rows argument needs to be higher than zero.")
+  }
+  if(!(sorting == "nodes" || sorting == "subjects")){
+    stop("Unsupported sorting type. Only nodes and subjects allowed")
+  }
+}
+
+applySorting <- function(df,sorting){
+  measurements <- df[,3:ncol(df)]
+  colNames <- names(measurements)
+
+  subsets <- getSubset(colNames)
+  nodes <- getNode(colNames)
+  subjects <- getSubject(colNames)
+  if(sorting == "nodes"){
+    colNames <- paste(subsets, nodes, subjects, sep="")
+  }else{
+    colNames <- paste(subsets, subjects, nodes, sep="")
+  }
+  inds <- sort(colNames, index.return=TRUE )$ix
+
+  measurements <- measurements[,inds]
+  cbind(df[,c(1,2)], measurements)
+}
+
+getNode <- function(patientIDs){
+  splittedIds <- strsplit(patientIDs,"_") # During merge, which is always run we append subset id, either _s1 or _s2 to PATIENTID.
+  sapply(splittedIds, FUN=tail_elem,n= 2) # In proper patienid subset will always be  at the end. This select last but one elemnt - the node
+}
+
+getSubject <- function(patientIDs){
+  splittedIds <- strsplit(patientIDs,"_")
+  sapply(splittedIds, FUN=tail_elem,n= 3)
+}
```

Comment:
If subject name/sample name contains "_" you will only retrieve a partial subject/sample name. Eg subject_123_n1_s1 will return "123". I am not sure if underscores are allowed in subject/sample names, but I think it may be allowed and therefore can occur